FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/tls_freertos.c

```diff
@@ -380,23 +450,16 @@ MbedtlsStatus_t Mbedtls_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
                     pNetworkContext,
                     pHostName,
                     pNetworkCredentials ) );
-        returnStatus = MBEDTLS_INVALID_PARAMETER;
+        returnStatus = TLS_TRANSPORT_INVALID_PARAMETER;
     }
-    else if( ( pNetworkCredentials->pRootCa == NULL ) ||
-             ( pNetworkCredentials->pClientCert == NULL ) ||
-             ( pNetworkCredentials->pPrivateKey == NULL ) )
+    else if( ( pNetworkCredentials->pRootCa == NULL ) )
```

Comment:
Why were NULL checks for the client cert and private key removed?

Reply:
For only server auth, it can be NULL.